src/Gateway.php

```diff
@@ -57,6 +57,27 @@ public function __construct() {
 		add_action( 'woocommerce_checkout_update_order_meta', array( $this, 'process_custom_checkout_fields' ) );
 	}
 
+	/**
+	 * Check if the gateway is the chosen payment method.
+	 *
+	 * Due to inconsistencies in the gateway name (e.g., ledyer_payments vs. ledyer_payments_invoice), we need to check for the presence of 'ledyer_payments'.
+	 *
+	 * @param int|null|\WC_Order $order_id The WooCommerce order or its id.
+	 * @return bool
+	 */
+	private function is_chosen_gateway( $order_id = null ) {
+		if ( $order_id instanceof \WC_Order ) {
+			$chosen_gateway = $order_id->get_payment_method();
+		} elseif ( ! empty( $order_id ) ) {
+			$order          = wc_get_order( $order_id );
+			$chosen_gateway = $order->get_payment_method();
```

Comment:
This row could cause a fatal error if the wc_get_order does not return an actual order.
I would also change the name of the variable to be `$order` instead, since checking for a payment method on a variable named `$order_id` can be missleading
I know for example with blocks, since the order might still be in draft status, the function wc_get_order wont return it, and instead return false as if it does not exist. So maybe do a change to the logic to be more like:
```php
if ( ! empty( $order ) && ! $order instanceof \WC_Order ) {
    $order = wc_get_order( $order )
}

if( $order ) {
    $chosen_gateway = $order->get_payment_method();
} else { 
    $chosen_gateway = WC()->session->get( 'chosen_payment_method' );
}

return strpos( $chosen_gateway, 'ledyer_payments' ) !== false;
```

Reply:
Probably should add a check as well if `WC()->session` is not null incase this is ever used on admin pages too.